tuf/tuf.go

```diff
@@ -185,10 +211,19 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey) error
 		return data.ErrInvalidRole{Role: role.Name, Reason: "not a valid delegated role"}
 	}
 	parent := filepath.Dir(role.Name)
-	p, ok := tr.Targets[parent]
-	if !ok {
+
+	// check the parent role
+	if parentRole := tr.keysDB.GetRole(parent); parentRole == nil {
 		return data.ErrInvalidRole{Role: role.Name, Reason: "parent role not found"}
 	}
+
+	// check the parent role's metadata
+	p, ok := tr.Targets[parent]
+	if !ok { // the parent targetfile may not exist yet - if not, then create it
+		p = data.NewTargets()
```

Comment:
@riyazdf brought up a great point about the assumption of `keysDB` being correct (we check `keysDB` to be sure the parent role exists first).  If it doesn't have the parent role, and should, then the repo may just be in a bad state, and we correctly error.  If it shouldn't have the parent role, but does though, that would let UpdateDelegations proceed.  I'm wondering if it would be too defensive to check if the grandparent metadata exists before adding the parent metadata?